kedro/framework/cli/starters.py

```diff
@@ -149,10 +184,26 @@ def starter():
 @starter.command("list")
 def list_starters():
     """List all official project starters available."""
-    repo_url = _STARTERS_REPO.replace("git+", "").replace(".git", "/tree/main/{alias}")
-    output = [
-        {alias: repo_url.format(alias=alias)} for alias in sorted(_STARTER_ALIASES)
-    ]
+    starters_aliases = _get_starters_aliases()
+
+    output = {}
+    for starter_name, starter_config in starters_aliases.items():
+        directory = starter_config.get("directory")
+        repo_url = starter_config.get("repo")
+        repo_url = repo_url.replace("git+", "")
+        if directory:
+            repo_url = repo_url.replace(".git", f"/tree/main/{directory}")
```

Comment:
Again, ideally I'd make this generic to work for things outside git.

Reply:
This is your original implementation (with only your starters) and I did not dare change it, but I agree. I've changed it to make it generic